catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java

```diff
@@ -204,6 +206,22 @@ QueryResponse doQuery(QueryRequest queryRequest, FederationStrategy strategy)
 
         LOGGER.debug("Calling strategy.federate()");
 
+        Query originalQuery = queryRequest.getQuery();
+
+        long timeout = originalQuery.getTimeoutMillis();
+
+        Query modifiedQuery = new QueryImpl(originalQuery,
```

Comment:
Why create new instances of Query and QueryRequest? If we really need to do this, can we make copy constructors on QueryRequest and Query?